### PR TITLE
Robustify

### PR DESCRIPTION
**Pull request by vincent-miller**
Updated Clamp to use Anna's suggestions, and fixed the motor direction. I changed the encoder direction as well, but I don't know if that is correct. The encoder direction also might not matter once min and max are set.

### CODE REVIEW

File: QuasarHelios2015/src/org/team1540/quasarhelios/Clamp.java [hunk 1]
```diff
@@ -16,13 +17,21 @@
...
 
 public class Clamp {
-    public static FloatStatus heightOrSpeed = new FloatStatus();
+    public static final boolean MODE_SPEED = true;
+    public static final boolean MODE_HEIGHT = false;
+    
+    public static FloatStatus height = new FloatStatus();
+    public static FloatStatus speed = new FloatStatus();
     public static BooleanStatus mode = new BooleanStatus(); // true = speed,
                                                             // false = height
```
Comment (AidanSmith) [inline QuasarHelios2015/src/org/team1540/quasarhelios/Clamp.java]: You can remove the comment now that you have fields for MODE_SPEED and MODE_HEIGHT.

File: QuasarHelios2015/src/org/team1540/quasarhelios/Clamp.java [hunk 2]
```diff
@@ -30,12 +39,12 @@
...
 
-        FloatInputPoll encoder = Igneous.makeEncoder(10, 11, false);
+        FloatInputPoll encoder = Igneous.makeEncoder(10, 11, true);
         ExtendedMotor clampCAN = Igneous.makeCANTalon(1);
         FloatOutput motorControlTemp = FloatMixing.ignoredFloatOutput;
-        
+
         try {
-            motorControlTemp = clampCAN.asMode(ExtendedMotor.OutputControlMode.VOLTAGE_FRACTIONAL);
+            motorControlTemp = FloatMixing.negate(clampCAN.asMode(ExtendedMotor.OutputControlMode.VOLTAGE_FRACTIONAL));
```
Comment (celskeggs) [inline QuasarHelios2015/src/org/team1540/quasarhelios/Clamp.java]: I already fixed this, IIRC. See the negation below in `speedControl`.

File: QuasarHelios2015/src/org/team1540/quasarhelios/DriveCode.java [hunk 3]
```diff
@@ -158,10 +158,11 @@ public static void setup() {
         fieldCentric.setTrueWhen(EventMixing.filterEvent(startFieldCentric, true, Igneous.startTele));
         octocanumShifting.toggleWhen(octocanumShiftingButton);
         FloatMixing.pumpWhen(octocanumShiftingButton, HeadingSensor.absoluteYaw, desiredAngle);
+
         Igneous.duringTele.send(EventMixing.filterEvent(octocanumShifting, true, mecanum));
         Igneous.duringTele.send(EventMixing.filterEvent(octocanumShifting, false,
                 DriverImpls.createTankDriverEvent(leftJoystickY, rightJoystickY, leftMotors, rightMotors)));
-
+
```
Comment (celskeggs) [inline QuasarHelios2015/src/org/team1540/quasarhelios/DriveCode.java]: ?

**Comment by celskeggs** [comment]
Just those.

**Comment by celskeggs** [comment]
Looks good.

**Pull request merged**